samples/subsys/logging/logger/src/main.c

```diff
@@ -172,7 +172,8 @@ static void performance_showcase(void)
 	u32_t window = 2;
 
 	do {
-		LOG_INF("performance test - log message %d", cnt++);
+		LOG_INF("performance test - log message %d", cnt);
```

Comment:
As @andrewboie proposed here: https://github.com/zephyrproject-rtos/zephyr/pull/8828#issuecomment-403934159, we could use static inline functions in order to not change behaviour when logger is enabled and disabled.

@nordic-krch; Have you tried this approach?

Reply:
macros gives me possibility to handle variable number of arguments without using va_list. but even when inline functions would be used I need to handle the case when some variables that macro/function access disappears when logger is disabled for given module.

Reply:
if the log is disabled you can just make empty variadic inline functions.
I really would rather see this with inline functions at some point in the layers so that variables defined specifically for use in a log point don't generate "defined but not used" warnings.

Reply:
This is ensured, if you look at macro __LOG() (log_core.h line 169) you can see that it always evaluates to *if else* clause. Condition in *if* clause evaluates to false on compile time if log is disabled for given module and log code is not included in the binary but you will never get 'defined but not used'.